### PR TITLE
Random Category blog and list order

### PR DESCRIPTION
**Pull request by brianteeman**
This PR answers a fairly common request on the forums to add a random ordering option for articles in the blog category and category list view.
### To test
1. create a new category blog menu item and in the blog layout options go to article order and you will see a new option “Random Order”.  Select this option and then check the article order on the front end. Refresh the page and the article order will have changed
2. create a new category list menu item and in the list layout options go to article order and you will see a new option “Random Order".  Select this option and then check the article order on the front end. Refresh the page and the article order will have changed

### CODE REVIEW

**Comment by brianteeman** [comment]
Just to add that as there are no changes to existing functionality there are no B/C issues<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7451">issues.joomla.org/joomla-cms/7451</a>.</sub>

File: components/com_content/helpers/query.php [hunk 1]
```diff
@@ -104,6 +104,10 @@ public static function orderbySecondary($orderby, $orderDate = 'created')
 			case 'front' :
 				$orderby = 'a.featured DESC, fp.ordering, ' . $queryDate . ' DESC ';
 				break;
+				
+			case 'random' :
+				$orderby ='RAND()';
```
Comment (smehrbrodt) [inline components/com_content/helpers/query.php]: There is no RAND function in Postgres. It's called [RANDOM](http://www.postgresql.org/docs/9.4/static/functions-math.html#FUNCTIONS-MATH-RANDOM-TABLE) there.

Comment (brianteeman) [inline components/com_content/helpers/query.php]: Dear. It's also different in mssql and isn't very good if you have a large
data set http://www.petefreitag.com/item/466.cfm

**Comment by alikon** [comment]
should be not too hard  to let the "random order" portable across supported db's
let me sumbit a "portabily" pr to this one ;)

**Comment by brianteeman** [comment]
Great

**Comment by brianteeman** [comment]
@alikon Sorry I missed your PR - is it just #5 you want

**Comment by brianteeman** [comment]
@alikon I should have read your pulls before commenting - i have applied them all - thanks

**Comment by brianteeman** [comment]
Sorry I have no idea how to fix the travis issues - i "think" they are unrelated to the PR and are pre-existing

**Comment by rdeutz** [comment]
the added function Rand needs a function comment, that's all, it is added by this PR so it is realted

**Comment by brianteeman** [comment]
@alikon did that not me so I hope he can do the comment

File: libraries/joomla/database/query/postgresql.php [hunk 2]
```diff
@@ -638,4 +638,8 @@ public function regexp($value)
 	{
 		return ' ~* ' . $value;
 	}
+	public function Rand()
+	{
+		return ' RAND() ';
```
Comment (smehrbrodt) [inline libraries/joomla/database/query/postgresql.php]: Still spelled wrongly. Should be RANDOM().

**Comment by alikon** [comment]
comment added and typo fixed

File: libraries/joomla/database/query/postgresql.php (hunk 2 above)
Comment (Fedik) [inline libraries/joomla/database/query/postgresql.php]: and I guess it need to be lowercased, as other methods in this class

**Comment by wojsmol** [comment]
@brianteeman I send you simple CS fix https://github.com/brianteeman/joomla-cms/pull/12

**Comment by yanivRozenman** [comment]
The test was successful<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7451">issues.joomla.org/joomla-cms/7451</a>.</sub>

**Comment by brianteeman** [comment]
Thanks for testing @Bodge-IT @yanivRozenman Setting RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7451">issues.joomla.org/joomla-cms/7451</a>.</sub>

**Comment by Bodge-IT** [comment]
Hi Brian,
Really chuffed to be able to contribute.  JoomlaLondon now up and running too, after we got over a small issue with the patch tester this week.

Will start more regular testing from here, both individuals and as a group at the monthly #JUGL meets.
Cheers
Gary

**Comment by brianteeman** [comment]
Great news.

On 22 July 2015 at 13:15, Gary Barclay <notifications@github.com> wrote:

> Hi Brian,
> Really chuffed to be able to contribute. JoomlaLondon now up and running
> too, after we got over a small issue with the patch tester this week.
>
> Will start more regular testing from here, both individuals and as a group
> at the monthly #JUGL meets.
> Cheers
> Gary
>
> From: Brian Teeman [mailto:notifications@github.com]
> Sent: 22 July 2015 10:45
> To: joomla/joomla-cms <joomla-cms@noreply.github.com>
> Cc: Gary Barclay <gary@barclayworks.com>
> Subject: Re: [joomla-cms] Random Category blog and list order (#7451)
>
>
> Thanks for testing @Bodge-IT<https://github.com/Bodge-IT> @yanivRozenman<
> https://github.com/yanivRozenman> Setting RTC
>
> ________________________________
> This comment was created with the J!Tracker Application<
> https://github.com/joomla/jissues> at issues.joomla.org/joomla-cms/7451<
> http://issues.joomla.org/tracker/joomla-cms/7451>.
>
> —
> Reply to this email directly or view it on GitHub<
> https://github.com/joomla/joomla-cms/pull/7451#issuecomment-123644899>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/7451#issuecomment-123699241>.
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by roland-d** [comment]
Thanks all, merged into 3.5-dev with commit d6f35c.

**Pull request closed without merging**